From 422b08d47dca630396fa38d4239b9c0847dd4fb5 Mon Sep 17 00:00:00 2001
From: Joe LeVeque <jolevequ@microsoft.com>
Date: Fri, 6 Dec 2019 06:09:17 +0000
Subject: [PATCH] Don't skip down interfaces when discovering interfaces in
 relay mode

When discovering interfaces in relay mode, don't skip interfaces just
because they're down. If we fail to discover the interfaces because they
are down when the relay starts, but then are brought up at a later point
in time, the relay will discard any packets received on them because it
didn't discover the interface(s) when it started up.
---
 common/discover.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/discover.c b/common/discover.c
index bf82735..f57e3ef 100644
--- a/common/discover.c
+++ b/common/discover.c
@@ -1014,7 +1014,8 @@ discover_interfaces(int state) {
 		      info.flags & IFF_LOOPBACK ||
 		      info.flags & IFF_POINTOPOINT) && !tmp) ||
 		    (!(info.flags & IFF_UP) &&
-		     state != DISCOVER_UNCONFIGURED))
+		     state != DISCOVER_UNCONFIGURED &&
+		     state != DISCOVER_RELAY))
 			continue;
 		
 		/* If there isn't already an interface by this name,
-- 
2.17.1

